scripts/prepare_release.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+[[ $1 =~ ^prepare-([0-9]\.[0-9]+)\.0$ ]] ||
+    { echo "Target format error: should be 'prepare-x.x.0', got '$1'"; exit 1; }
+
+VERSION="${BASH_REMATCH[1]}"
+
+git checkout "${MASTER}"
+git pull --ff-only "${ISTIOIO_GIT_SOURCE}" "${MASTER}"
+
+sed -i "s/^source_branch_name: .*$/source_branch_name: release-${VERSION}/" data/args.yml
```

Comment:
IIUC the purpose of this script is to update the istio/istio.io master branch to use a release branch of istio/istio?

Why would we not simply create a matching branch of istio/istio.io on the same day as the istio/istio branch is cut?  Then we would immediately have istio.io/release-x.y.z pointing to istio/release-x.y.z and istio.io/master could still be pointing at istio/master (i.e. the bleeding edge).

Otherwise, we lose track of istio/master WRT docs for the lifetime of the release. If someone were to break docs in master (not in the istio/release-x.y.z branch), we wouldn't know until the release is completed.

@davidhauck any thoughts?

Reply:
Ah I see. I guess our goal should be to ultimately change the process so that documentation work is done earlier in the release cycle. If that were the case, then there wouldn't be the mad flurry of docs changes at the end ... which would probably make everyone on the docs team a lot happier anyway :).

But since we're not there yet, I guess this is the lesser of evils.